pkg/apis/networking/types.go

```diff
@@ -583,3 +583,62 @@ type ServiceBackendPort struct {
 	// +optional
 	Number int32
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ClusterCIDRConfig is the Schema for the clustercidrconfigs API.
+type ClusterCIDRConfig struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+
+	Spec   ClusterCIDRConfigSpec
+	Status ClusterCIDRConfigStatus
+}
+
+// ClusterCIDRConfigSpec defines the desired state of ClusterCIDRConfig.
+type ClusterCIDRConfigSpec struct {
+	// NodeSelector defines which nodes the config is applicable to.
+	// An empty or nil NodeSelector functions as a default that applies to all nodes.
+	// This field is immutable.
```

Comment:
Immutable fields are only meaningful if there is a finalizer. Does this resource attach a finalizer if there are allocations from the given CIDR?

Reply:
Yes it does attach a finalizer during creation https://github.com/sarveshr7/kubernetes/blob/42990f843a804e395ea6c98d49098b7a8def8c25/pkg/controller/nodeipam/ipam/multi_cidr_range_allocator.go#L998-L1000